test/Quaternion.jl

```diff
@@ -571,4 +571,13 @@ end
             @test_throws DivideError lyap(null, null)
         end
     end
+
+    @testset "RealDot with $T" for T in (Float32, Float64)
+        for _ in 1:10
+            q1 = randn(Quaternion{T})
+            q2 = randn(Quaternion{T})
+            @test real(dot(q1,q2)) == @inferred(realdot(q1,q2))
```

Comment:
Can you also check the other 3 permutations? Something like this
```suggestion
            @test @inferred(realdot(q1,q2)) == realdot(q2, q1) == real(dot(q1, q2)) == real(dot(q2, q1))
```

Reply:
Since checking `real(dot(q1, q2)) == real(dot(q2, q1))` does not seem to be our responsibility, I have added another commutative law test `@test realdot(q1,q2) == realdot(q2,q1)`. Is this okay?

Reply:
Yes, that's fine, but I think we also want to check if one of those is equal to `real(dot(q2, q1))`.

Reply:
I added the test in https://github.com/JuliaGeometry/Quaternions.jl/pull/119/commits/204e32079a8512db0d022a0f4b5706cc08df898e 👍